client/src/utils/responsiveness.ts

```diff
@@ -0,0 +1,54 @@
+import { useState, useEffect } from 'react'
+
+/**
+ * Given a CSS media query, returns whether the media query is matched, e.g.
+ *
+ * const isDesktopWidth = useMediaQuery('(min-width: 64em)')
+ *
+ * Listens for updates, e.g. in response to window resizing.
+ */
+export function useMediaQuery(query: string): boolean {
+  const [isMatched, setIsMatched] = useState(false)
+
+  useEffect(() => {
+    const matchMedia = window.matchMedia(query)
+    setIsMatched(matchMedia.matches)
```

Comment:
To the first q, this is an unlikely scenario but say you have the following code:

```ts
const useAlternateLayout = useMediaQuery(useLowerThreshold
  ? '(min-width: 400px)'
  : '(min-width: 800px)'
)
```

If `useLowerThreshold` were toggled from `false` to `true`, the query provided to `useMediaQuery` would change. And if the "initial" value were set outside the `useEffect`, the hook wouldn't consider the updated query until the screen was (unrelatedly) resized.

We could additionally set the initial value in the `useState` invocation, but we'd still need the initial value in the `useEffect` for the case outlined above.

To the second q, agreed! Will update

Reply:
Gotcha, that makes sense!

Reply:
Actually thought about the second suggestion some more and settled on something a bit different to ensure that the proper values are set before the DOM is painted.

The operations performed inside the effect are asynchronous (updating state values, setting up listeners), so changing from `useEffect` to `useLayoutEffect` wouldn't make a difference. I instead updated the `useMediaQuery` hook to set its initial state value via a call to `window.matchMedia` since the first values returned by the `useMediaQuery` hook do affect the first paint.

https://github.com/votingworks/arlo/pull/1608/commits/f798919b7878c5509b2945ff05ba81bbf9da93bc